pwiz_tools/Shared/ProteowizardWrapper/MsDataFileImpl.cs

```diff
@@ -1036,7 +1036,7 @@ private double[] GetIonMobilityArray(Spectrum s)
 
         private BinaryDataDouble TryGetIonMobilityData(Spectrum s, CVID cvid, ref CVID? cvidIonMobility)
         {
-            using var data = s.getArrayByCVID(cvid)?.data;
+            var data = s.getArrayByCVID(cvid)?.data;

```

Comment:
I think "TryGetIonMobilityData" should be changed so that it returns a double[] like this:
```
        private double[] TryGetIonMobilityData(Spectrum s, CVID cvid, ref CVID? cvidIonMobility)
        {
            using var data = s.getArrayByCVID(cvid)?.data;
            if (data != null)
                cvidIonMobility = cvid;

            return data?.Storage();
        }
```
I am a little confused, though, as to why the current code works and doesn't crash. It might be that "BinaryDataDouble" does not actually do anything.

Reply:
Yeah, this was a "why did that ever work?" thing for sure. 

@chambm what do you think of Nick's suggestion?

Reply:
It looks like the reason that the original did not crash is that "data.owner_" is always non-null, and the destructor for "BinaryData" does not do anything in that case:
`    virtual ~BinaryData() { if (owner_ == nullptr && base_ != NULL) delete base_; }
`

Reply:
It's worth trying as double[]. None of the vendors are providing ion mobility arrays in a managed context, so Storage() will always copy the array from native memory to managed. I'm not sure whether that copying would be worse than iterating over the native memory, so definitely compare the IMS perftests before and after. (But certainly that using declaration was wrong even if it was a no-op.)

Reply:
I don't think there can be any performance impact from this change. The caller ("GetIonMobilityArray") used to always immediately call "Storage()" on what was being returned by this method. This change just makes it so that the call to "Dispose()" happens after instead of before the call to "Storage()".

Reply:
I read this the same way as Nick. I think it's good to go, thanks all.